services/API-service/src/api/data/data.service.ts

```diff
@@ -124,40 +117,25 @@ export class DataService {
     return result[0];
   }
 
-  public async getTriggeredAreas(event: number): Promise<TriggeredArea[]> {
-    const query =
-      'select pcode,name,population_affected \
-    from "IBF-pipeline-output".event_districts \
-    where event = $1 \
-    order by population_affected DESC';
-
-    const result = await this.manager.query(query, [event]);
+  public async getTriggeredAreas(
+    countryCode: string,
+  ): Promise<TriggeredArea[]> {
+    const query = fs
+      .readFileSync('./src/api/data/sql/get-triggered-areas.sql')
+      .toString();
 
+    const result = await this.manager.query(query, [countryCode, countryCode]);
     return result;
   }
 
-  public async getEvent(countryCode: string): Promise<DisasterEvent> {
-    const daysStickyAfterEvent = 0;
-
-    const query =
-      "select t0.* \
-        from \"IBF-pipeline-output\".events t0 \
-        left join( \
-          select max(case when end_date is null then '9999-99-99' else end_date end) as max_date \
-          , country_code \
-      from \"IBF-pipeline-output\".events \
-      group by country_code \
-        ) t1 \
-        on t0.country_code = t1.country_code \
-        where t0.country_code = $1 \
-        and(case when t0.end_date is null then '9999-99-99' else end_date end) = t1.max_date \
-        and(end_date is null or to_date(end_date, 'yyyy-mm-dd') >= current_date - " +
-      daysStickyAfterEvent +
-      ') \
-    ';
-
+  public async getCountryEvent(countryCode: string): Promise<CountryEvent> {
+    const query = fs
+      .readFileSync('./src/api/data/sql/get-country-event.sql')
+      .toString();
     const result = await this.manager.query(query, [countryCode]);
-
+    if (!result[0].country_code) {
```

Comment:
Better to find the source of invalid `country_code` and fix it there. Probably a `nullable` column.

Reply:
Ah the idea here is that the get-country-event query uses max and it therefore always return a row even if there is no event:
![image](https://user-images.githubusercontent.com/34537157/109429335-90525880-79fb-11eb-852e-99d3f1a9e9c2.png)
So this is a trick to check whether the is an active placeCode event in a country. I changed it to (!result[0].start_date) to make it clearer